### DISCUSSION TITLE
`FileType` and `BufEnter` event not emitted for `help`

### DISCUSSION THREAD

**Discussion by simonmandlik** [neovim/neovim#31518] [General]
Not sure if I'm doing something wrong or if this is a bug. [This](https://stackoverflow.com/questions/37804435/how-to-enable-line-numbers-for-the-vim-help-permanently) is the recommended way to achieve numbering in help. But I have noticed that closing the help window and later running the same `:h` command will lead to the event not firing (even for `BufEnter`). When opening a different help page, the event is emitted:

https://github.com/user-attachments/assets/a66ac997-4178-4ccd-a5ae-52214360ab76

`vimr`:

```vimscript
au FileType help setlocal number
au BufEnter help setlocal number
```

**Comment by zeertzjq** [comment]
It seems to work if you add `bufhidden=unload`:
```vim
autocmd FileType help setlocal number bufhidden=unload
```

**Comment by wsdjeg** [comment]
Because number is local to window, and not local to buffer.

1. `:h help` will open `helphelp.txt` buffer, and filetype is set, then FileType event is called, number is set to true
2. `:helpclose` will close the help win, and the local number opt is cleared too. and the current win number is false.
3. then `:h help` will open new window, the number opt is copyed from current win, is false too.
4. `BufEnter`'s pattarn is bufname, not filetype, so `BufEnter helphelp.txt`  works for `:h help`, That means: This works for `:h help`
 `helpclose` `:h help`
     ```
    au FileType help setlocal number
    au BufEnter helphelp.txt setlocal number
     ```

if you do not set bufhidden option, maybe you can use

```
au BufWinEnter * if &ft == 'help' | setlocal number | endif
```
